server/services/teamsSummaryCountService.ts

```diff
@@ -0,0 +1,154 @@
+import logger from '../../logger'
+import type { StrapiApiClient, RestClientBuilder } from '../data'
+import { Component, DataItem, Product, SingleResponse } from '../data/strapiApiTypes'
+import AlertsService from './alertsService'
+
+type TeamAlertSummary = {
+  products: Record<string, Record<string, number>>
+  total: number
+}
+
+export default class TeamsSummaryCountService {
+  constructor(
+    private readonly alertsService: AlertsService,
+    private readonly strapiClient: RestClientBuilder<StrapiApiClient>,
+  ) {}
+
+  /**
+   * Helper: Fetch all products for a team by team slug
+   */
+  async getProductsForTeam(teamSlug: string): Promise<DataItem<Product>[]> {
+    try {
+      const team = await this.strapiClient('').getTeam({ teamSlug })
+      const products = Array.isArray(team.data)
+        ? team.data[0]?.attributes?.products?.data || []
+        : team.data?.attributes?.products?.data || []
```

Comment:
I think what @andrewrlee is doing fixes this but otherwise can be pulled into a function, something like:

```
const getData = data => {
  return Array.isArray(data.data)
    ? data.data[0]?.attributes?.products?.data || []
    : data.data?.attributes?.products?.data || []
}
```
the `products` part could be passed as an arg maybe?

Then can just do `const products = getData(team)` ?